queueRetention/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import config from "./config";
+import { prisma } from "./prismaClient";
+
+export async function redactSubmissions() {
+  const retentionAsTimestamp =
+    parseInt(config.retentionPeriod) * 24 * 60 * 60 * 1000;
+  const retentionLimit = new Date().getTime() - retentionAsTimestamp;
+  const validSubs = await prisma.submission.findMany({
+    where: {
+      updated_at: {
+        lt: retentionLimit,
+      },
+      complete: true,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (validSubs.length > 0) {
+    prisma.submission.deleteMany({
+      where: {
+        id: {
+          in: validSubs.map((sub) => sub.id),
+        },
+      },
+    });
+  }
+}
+
+(async () => {
```

Comment:
I would not iife this.

1. create a file which matches the name of the function
2. in this index file
```.ts
// index.ts
import { redactSubmissions } from "./redactSubmissions" 


async function main() {
  // log some starting scripts   

  const deleted = await redactSubmissions();
  
  // log deleted or log and handle if anything failed

}

main()
  .then(() => {
    logger.info("All tasks completed");
    process.exit(0); // <- tells systems, including kube that something exited doing the task successfully 
  })
  .catch((promiseRejects) => {
    logger.error(promiseRejects);
    process.exit(1); // // <- tells systems, including kube that something failed. for exit code 1 - depending on kube config, it may try to restart the process immediately
  });

```